### DISCUSSION TITLE
Multiple Coder instances with one database?

### DISCUSSION THREAD

**Discussion by kotx** [coder/coder#3613] [General] [answered]
Currently hosting two Coder instances that are connected to the same database seems to work, but sees a huge performance decrease. When I change the state of a workspace, each line is very slow to arrive, unlike when I had only one instance. 
1) Are running multiple instances on one database supported?
2) If so, how can I prevent this locking-up behaviour?

My use case for this is to have Coder instances in multiple regions, so proxied requests to workspaces don't need to round-trip. I'm aware that the database is still in one location and thus there will be some latency, but at least not while proxying code-server/ssh/other ports.

**Comment by johnstcn** [comment, marked as answer]
Hi @kotx, [high availability support](https://github.com/coder/coder/issues/3227) is something we're actively working on!
